Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -6514,8 +6514,14 @@ Frame* WebPage::mainFrame() const
     return m_page ? &m_page->mainFrame() : nullptr;
 }
 
-// FIXME: This should return an FrameView.
-LocalFrameView* WebPage::mainFrameView() const
+FrameView* WebPage::mainFrameView() const
+{
+    if (auto* frame = mainFrame())
+        return frame->virtualView();
+    return nullptr;
+}
+
+LocalFrameView* WebPage::localMainFrameView() const
 {
     if (auto* frame = dynamicDowncast<LocalFrame>(mainFrame()))
```

Comment:
This could just return dynamicDowncast<LocalFrameView>(mainFrameView())
That would reduce places in the code where we are trying to do basically the same thing with code that looks a little different.